lib/hyperswitch.js

```diff
@@ -54,7 +54,12 @@ function HyperSwitch(options, req, parOptions) {
         this._rootReq.headers = this._rootReq.headers || {};
         this._requestFilters = par._requestFilters;
         this._subRequestFilters = par._subRequestFilters;
-        this.ctx = par.ctx || {};
+        this.ctx = par.ctx || {
+            headers: {
+                'user-agent': req.headers['user-agent'],
+                'x-client-ip': req.headers['x-client-ip'],
```

Comment:
May be it's better to forward `X-Forwarded-For`? It already contains the `X-Client-IP` as the last element of the list, but provides more information, and it's designed to be forwarded.